substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/SubstrateOptions.java

```diff
@@ -469,4 +469,14 @@ protected void onValueUpdate(EconomicMap<OptionKey<?>, Object> values, Integer o
     @Option(help = "Search path for source files for Application or GraalVM classes (list of comma-separated directories or jar files)")//
     public static final HostedOptionKey<String[]> DebugInfoSourceSearchPath = new HostedOptionKey<String[]>(null) {
     };
+    @Option(help = "Directory under which to create source file cache for Application or GraalVM classes")//
+    public static final HostedOptionKey<String> DebugInfoSourceCacheRoot = new HostedOptionKey<String>("sources") {
+        @Override
+        protected void onValueUpdate(EconomicMap<OptionKey<?>, Object> values, String oldValue, String newValue) {
+            // disallow null or empty string
+            if (newValue == null || newValue.length() == 0) {
```

Comment:
It's better to issue an error message here instead of silently fixing a users bad option argument. Otherwise we are breeding such command line invocations.

Reply:
Ok. What would be the appropriate way to make the user aware of the error? Do I throw an exception to stop the native image run? Should Can log an error message and force an exit? Or do I log a warning message, reset the value to the default and carry on. Is there a standard API to implement the latter actions?

Reply:
See e.g. `NativeImageOptions.getCStandard()` on how we usually do this.

Reply:
Ok, thanks. Got it.

Reply:
done